### PR TITLE
Add optional non-global timeout parameter to functions that use timeouts

### PR DESCRIPTION
**Pull request by mattremmel**

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
The code looks good to me but it would be nice to see a test or two demonstrating that it works.

File: RZBluetooth/Command/RZBCommandDispatch.h [hunk 1]
```diff
@@ -89,6 +89,11 @@
  */
 - (void)dispatchCommand:(RZBCommand *)command;
 
+/**
+ * Submit a new command with custom timeout.
+ */
+- (void)dispatchCommand:(RZBCommand *)command timeout:(NSTimeInterval) timeout;
```
Comment (joshbrown-eg) [inline RZBluetooth/Command/RZBCommandDispatch.h]: Spacing is inconsistent with the rest of the parameters in this file.

```suggestion
- (void)dispatchCommand:(RZBCommand *)command timeout:(NSTimeInterval)timeout;
```

File: RZBluetooth/Command/RZBCommandDispatch.m [hunk 2]
```diff
@@ -38,6 +38,12 @@ - (NSArray *)synchronizedCommandsMatching:(NSPredicate *)predicate
 }
 
 - (void)dispatchCommand:(RZBCommand *)command
+{
+    NSTimeInterval timeout = [RZBUserInteraction timeout];
+    [self dispatchCommand:command timeout:timeout];
+}
+
+- (void)dispatchCommand:(RZBCommand *)command timeout:(NSTimeInterval) timeout
```
Comment (joshbrown-eg) [inline RZBluetooth/Command/RZBCommandDispatch.m]: Spacing.

```suggestion
- (void)dispatchCommand:(RZBCommand *)command timeout:(NSTimeInterval)timeout
```

**Review by joshbrown-eg — CHANGES_REQUESTED** [review]
The rest of this looks good. Agreed with Matt that tests would be nice.

File: RZBluetooth/Command/RZBCommandDispatch.m (hunk 2 above)
Comment (joshuatbrown) [inline RZBluetooth/Command/RZBCommandDispatch.m]:
```suggestion
- (void)dispatchCommand:(RZBCommand *)command timeout:(NSTimeInterval)timeout
```

Spacing is still off here. Proposing the fix.

File: RZBluetoothTests/RZBCommandDispatchTests.m [hunk 3]
```diff
@@ -161,4 +162,23 @@ - (void)testThreadSafety
...
+
+-(void)testCustomTimeout
+{
+    NSTimeInterval timeout = 10;
+    self.dispatch = [[RZBCommandDispatch alloc] initWithQueue:nil context:self];
+    RZBCTestCommand *cCmd = [[RZBCTestCommand alloc] initWithUUIDPath:RZBUUIDPath.cUUIDPath];
+    [RZBUserInteraction setEnabled:YES];
+    [self.dispatch dispatchCommand:cCmd timeout:timeout];
+    XCTAssertEqualWithAccuracy(cCmd.expiresAt, [[NSDate date] timeIntervalSinceReferenceDate] + timeout, 0.0001);
+}
```
Comment (joshuatbrown) [inline RZBluetoothTests/RZBCommandDispatchTests.m:182]: Tests look good. I'll run them to verify that they pass for me.

**Review by joshuatbrown — CHANGES_REQUESTED** [review]

**Comment by joshuatbrown** [comment]
OK, so it turns out I'd have to add another remote in order to check out your branch and run the tests, so I'm going to just trust that you've already run them and they pass. Next time it'd be great if you can create a branch in the `egineering-llc` repo.

**Review by joshuatbrown — APPROVED** [review]

**Comment by mattremmel** [comment]
@joshuatbrown I'm happy to do it in `egineering-llc` next time. I'm not sure i've got permissions to do that though. Or at least didn't when I started this.

**Comment by joshbrown-eg** [comment]
@mattremmel Yeah, you didn't used to have write access but I granted it sometime in the last hour, so I think you should be able to next time.

**Review by ghost — APPROVED** [review]

**Review by joshbrown-eg — APPROVED** [review]

**Pull request merged**